packages/angular_devkit/core/src/virtual-fs/host/sync.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { Observable } from 'rxjs/Observable';
+import { BaseException } from '../../exception/exception';
+import { Path, PathFragment } from '../path';
+import {
+  FileBuffer,
+  Host,
+  HostCapabilities,
+  HostWatchEvent,
+  HostWatchOptions,
+  Stats,
+} from './interface';
+
+
+export class SynchronousDelegateExpectedException extends BaseException {
+  constructor() { super(`Expected a synchronous delegate but got an asynchronous one.`); }
+}
+
+/**
+ * Implement a synchronous-only host interface (remove the Observable parts).
+ */
+export class SyncDelegateHost<T extends object> {
+  constructor(protected _delegate: Host<T>) {
+    if (!_delegate.capabilities.synchronous) {
+      throw new SynchronousDelegateExpectedException();
+    }
+  }
+
+  protected _doSyncCall<ResultT>(observable: Observable<ResultT>): ResultT {
```

Comment:
Not that this is necessarily bad but it seems like there's a decent amount of work going on here to get the result (sync call -> result -> observable -> _unwrap_ -> sync result).  I suppose that's unavoidable without a complete async/sync bifurcation.